core/src/main/scala/akka/persistence/cassandra/query/CassandraReadJournalConfig.scala

```diff
@@ -24,9 +25,82 @@ import scala.concurrent.duration._
 
   sealed trait Period
   case object Max extends Period
-  case class Fixed(duration: Duration) extends Period
+  case class Fixed(duration: FiniteDuration) extends Period
+
+  /**
+   * Stores and validates the back track configuration.
+   *
+   * Back tacking can not go further back and the persistence id clean up timeout as it would deliver
+   * previously delivered events for which the metadata has been dropped.
+   * This is validated at start up.
+   *
+   * Back tracking can not go further back than the start of the previous bucket as missing searches do not
+   * go back any further than that. The duration to the start of the previous bucket is constantly changing
+   * with time so it is passed into the period methods by the events by tag stage.
+   *
+   * @param metadataCleanupInterval How often metadata about persistence ids is cleanled up
+   * @param interval How often to back track
+   * @param period How far to back track
+   * @param longInterval A less frequent back track and can be used to go back further
+   * @param longPeriod How far to back track for the long interval
+   */
+  case class BackTrackConfig(
+      metadataCleanupInterval: Option[FiniteDuration],
+      interval: Option[FiniteDuration],
+      period: Period,
+      longInterval: Option[FiniteDuration],
+      longPeriod: Period) {
+
+    validatePeriodLengths()
+    validateIntervals()
+
+    private def validateIntervals() = {
+      (interval, longInterval) match {
+        case (None, Some(_)) =>
+          throw new IllegalArgumentException(
+            "interval must be enabled to use long-interval. If you want a single interval just set interval, not long-interval")
+        case (Some(i), Some(li)) => require(i < li, "interval must be smaller than long-interval")
+        case _                   =>
+      }
+    }
+    private def validatePeriodLengths() = {
+      def cleanUpGreaterThanPeriod(period: Period, name: String): Unit = {
+        (metadataCleanupInterval, period) match {
+          case (Some(cleanup), Fixed(p)) =>
+            require(cleanup > p, s"$name can not be greater than cleanup-old-persistence-ids")
```

Comment:
given that there could be a few millis clock skew between nodes, should we have some margin?

Reply:
I think that is a good idea. Mayeb 10% even